olp-cpp-sdk-core/include/olp/core/http/Network.h

```diff
@@ -83,5 +83,10 @@ class CORE_API Network {
   virtual void Cancel(RequestId id) = 0;
 };
 
+/**
+ * @brief Create default Network implementation.
+ */
+CORE_API std::unique_ptr<Network> CreateDefaultNetwork();
```

Comment:
What is the point of this still?
You can do it same as it is done now in NetworkProtocol GetProtocol():

```cpp
#ifdef NETWORK_HAS_CURL
  return std::make_shared<NetworkProtocolCurl>();
#endif
#ifdef NETWORK_HAS_ANDROID
  return std::make_shared<NetworkProtocolAndroid>();
#endif
#ifdef NETWORK_HAS_IOS
  return std::make_shared<NetworkProtocolIos>();
#endif
#ifdef NETWORK_HAS_WINHTTP
  return std::make_shared<NetworkProtocolWinHttp>();
#endif
  return nullptr;
```

Reply:
@andescu 

If every network implementation has its own CreteDefaultNetwork() you can plug & unplug network implementations solely in CMakeLists and let the linker do the job for you. There will be less subjects to change and make mistake in if you ever need to touch network implementations.

Reply:
Looks very confusing to me, having multiple declarations of the same method.
This also will make life very hard for those that want to implement own Network handler as they will also need to implement CreateDefaultNetwork() which will be very hard to achieve if you build olp sdk c++ as shared lib before.